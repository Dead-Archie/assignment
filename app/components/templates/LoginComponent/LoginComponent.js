import React from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import withStyles from '../../../lib/withStyles';
import styles from './LoginComponent.style';

class LoginComponent extends React.PureComponent {
  constructor(props) {
    super(props);
    this.email = React.createRef();
    this.password = React.createRef();

    this.state = {
      isRegistered: true,
    };
  }

  checkValidUser = () => {
    const { isLoggedin } = this.props;
    const { value: email } = this.email.current;
    const { value: password } = this.password.current;
    if (email && password) {
      const storedEmail = localStorage.getItem('email', email);
      const storedPassword = localStorage.getItem('password', password);
      if (email === storedEmail && password === storedPassword) {
        isLoggedin(true);
        Router.push(`/dashboard`);
      } else {
        alert('invalid Credentials');
      }
    } else {
      alert('email & password is mandetory');
    }
  };

  clearData = () => {
    this.setState({ isRegistered: false });
    this.email.current.value = '';
    this.password.current.value = '';
  };

  registerData = () => {
    const { value: email } = this.email.current;
    const { value: password } = this.password.current;
    if (email && password) {
      localStorage.setItem('email', email);
      localStorage.setItem('password', password);
      alert(' You are sucessfuly Registered, Please login now.');
      this.setState({ isRegistered: true });
      this.email.current.value = '';
      this.password.current.value = '';
    } else {
      alert('email & password is mandetory');
    }
  };

  render() {
    const { className } = this.props;
    const { isRegistered } = this.state;
    return (
      <div className={className}>
        <div className="container-login100">
          <div className="wrap-login100">
            <div className="wrap-input100 ">
              <input
                className="input100"
                placeholder="Email"
                type="text"
                name="email"
                ref={this.email}
              />
            </div>

            <div className="wrap-input100 ">
              <input
                className="input100"
                placeholder="Password "
                type="password"
                name="pass"
                ref={this.password}
              />
            </div>
            {isRegistered && (
              <div>
                <div className="container-login100-form-btn">
                  <div className="wrap-login100-form-btn">
                    <div className="login100-form-bgbtn" />
                    <button className="login100-form-btn" onClick={() => this.checkValidUser()}>
                      Login
                    </button>
                  </div>
                </div>

                <div className="text-center p-t-115">
                  <span className="txt1">Donâ€™t have an account?</span>

                  <button className="txt2" onClick={() => this.clearData()}>
                    Sign Up
                  </button>
                </div>
              </div>
            )}
            {!isRegistered && (
              <div className="container-login100-form-btn">
                <div className="wrap-login100-form-btn">
                  <div className="login100-form-bgbtn" />
                  <button className="login100-form-btn" onClick={() => this.registerData()}>
                    Register
                  </button>
                </div>
              </div>
            )}
          </div>
        </div>
      </div>
    );
  }
}

LoginComponent.propTypes = {
  className: PropTypes.string,
  isLoggedin: PropTypes.func,
};
LoginComponent.defaultProps = {
  className: '',
  isLoggedin: () => {},
};

export default withStyles(LoginComponent, styles);
export { LoginComponent };
