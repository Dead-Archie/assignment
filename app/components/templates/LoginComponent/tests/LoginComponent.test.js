import React from 'react';
import { shallow } from 'enzyme';

import { LoginComponent } from '../LoginComponent';

jest.mock('next/router', () => ({ push: () => {} }));

const sampleProps = {
  className: 'withStyles-sc-4wveuc-0 lgzevM',
  isLoggedin: () => {},
};

const setUp = props => {
  return shallow(<LoginComponent {...props} />);
};

describe('<LoginComponent />', () => {
  let LoginComponentTest;
  beforeEach(() => {
    LoginComponentTest = setUp(sampleProps);
  });

  test('should render correctly', () => {
    expect(LoginComponentTest).toMatchSnapshot();
  });
  test('check state', () => {
    const wrapper = setUp();
    const checkState = wrapper.state('isRegistered');
    expect(checkState).toEqual(true);
  });
  test('email ref define  ', () => {
    const wrapper = setUp();
    expect(wrapper.instance().email).toEqual({ current: null });
  });
  test('password ref define  ', () => {
    const wrapper = setUp();
    expect(wrapper.instance().password).toEqual({ current: null });
  });

  test('onclick Button : login btn', () => {
    const wrapper = setUp();
    const LoginBtn = wrapper.find('.login-btn');

    wrapper.instance().email = {
      current: {
        value: null,
      },
    };
    wrapper.instance().password = {
      current: {
        value: null,
      },
    };
    LoginBtn.simulate('click');
    expect(wrapper.instance().email).toBeTruthy();
    expect(wrapper.instance().password).toBeTruthy();
  });

  test('onclick Button : registration btn', () => {
    const wrapper = setUp();

    wrapper.setState({ isRegistered: false });
    const registrationButton = wrapper.find('.registration-btn');
    expect(registrationButton).toHaveLength(1);
    wrapper.instance().email = {
      current: {
        value: null,
      },
    };
    wrapper.instance().password = {
      current: {
        value: null,
      },
    };
    registrationButton.simulate('click');
    expect(wrapper.instance().email).toBeTruthy();
    expect(wrapper.instance().password).toBeTruthy();
  });

  test('onclick Button : clear Data', () => {
    const wrapper = setUp();
    const clearData = wrapper.find('.txt2');

    wrapper.instance().email = {
      current: {
        value: null,
      },
    };
    wrapper.instance().password = {
      current: {
        value: null,
      },
    };
    clearData.simulate('click');
    expect(wrapper.instance().email).toBeTruthy();
    expect(wrapper.instance().password).toBeTruthy();
  });
});
