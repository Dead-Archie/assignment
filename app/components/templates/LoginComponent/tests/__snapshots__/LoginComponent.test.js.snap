// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginComponent /> should render correctly 1`] = `
<div
  className="withStyles-sc-4wveuc-0 lgzevM"
>
  <div
    className="container-login100"
  >
    <div
      className="wrap-login100"
    >
      <div
        className="wrap-input100 "
      >
        <input
          className="input100"
          name="email"
          placeholder="Email"
          type="text"
        />
      </div>
      <div
        className="wrap-input100 "
      >
        <input
          className="input100"
          name="pass"
          placeholder="Password "
          type="password"
        />
      </div>
      <div>
        <div
          className="container-login100-form-btn"
        >
          <div
            className="wrap-login100-form-btn"
          >
            <div
              className="login100-form-bgbtn"
            />
            <button
              className="login100-form-btn login-btn"
              onClick={[Function]}
            >
              Login
            </button>
          </div>
        </div>
        <div
          className="text-center p-t-115"
        >
          <span
            className="txt1"
          >
            Donâ€™t have an account?
          </span>
          <button
            className="txt2"
            onClick={[Function]}
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
